### PR TITLE
Fix packaging with maven for Angular1

### PR DESCRIPTION
**Pull request by pascalgrimaud**
Related to my comment https://github.com/jhipster/generator-jhipster/issues/4549#issuecomment-266231078

Plz check closely my modifications before merging.

For angular1:
- avoid npm/yarn install, bower install, and gulp install when using dev profile
- missing bower install, and gulp build+test for prod profile

### CODE REVIEW

**Comment by deepu105** [comment]
@jdubois can you take a look at this please?

File: generators/server/templates/_pom.xml [hunk 1]
```diff
@@ -1203,37 +1203,12 @@
             </dependencies>
             <build>
                 <plugins>
-                    <%_ if (!skipClient) { _%>
+                    <%_ if (!skipClient && clientFw === 'angular2') { _%>
```
Comment (jdubois) [inline generators/server/templates/_pom.xml]: Why do we have this new `clientFw` variable? There is already an `angularVersion` variable

Comment (pascalgrimaud) [inline generators/server/templates/_pom.xml]: It has been changed by @deepu105

Comment (jdubois) [inline generators/server/templates/_pom.xml]: Oh, OK, it's indeed a better name, it doesn't depend on Angular

Comment (pascalgrimaud) [inline generators/server/templates/_pom.xml]: probably to make the possibility to use another thing than angular in the future

Comment (deepu105) [inline generators/server/templates/_pom.xml]: yes exactly :)

File: generators/server/templates/_pom.xml [hunk 2]
```diff
@@ -1203,37 +1203,12 @@
...
                     <plugin>
                         <groupId>com.github.eirslett</groupId>
                         <artifactId>frontend-maven-plugin</artifactId>
                         <version>${frontend-maven-plugin.version}</version>
                         <executions>
-                            <%_ if (yarnInstall) { _%>
```
Comment (deepu105) [inline generators/server/templates/_pom.xml]: @pascalgrimaud does this mean we wont have yarn support when using ng2 in dev?

Comment (pascalgrimaud) [inline generators/server/templates/_pom.xml]: @deepu105 : maybe I drop too much code here! We should support yarn too for ng2 of course, but I forget to ask you this question: in dev profile, is it mandatory to npm install / yarn for ng2 ?

Comment (jdubois) [inline generators/server/templates/_pom.xml]: For me we should be able to use one or the other, in both AngularJS 1 and Angular 2. Then, in the future, if Yarn works well enough, I would concentrate on Yarn.

**Comment by deepu105** [comment]
Actually given that you need to run webpack in order to do any development
we need not actually do everything from maven for dev we could just trigger
a webpack build post npm install like we used to do for gulp. let me take
another look at that tomorrow

Thanks & regards,
Deepu

**Comment by pascalgrimaud** [comment]
Just updated my PR :
- yarn support when using ng2 in dev
- add properties for node.version npm.version yarn.version

**Pull request merged**